seatunnel-engine/seatunnel-engine-server/src/main/java/org/apache/seatunnel/engine/server/task/SinkAggregatedCommitterTask.java

```diff
@@ -262,7 +263,12 @@ public void restoreState(List<ActionSubtaskState> actionStateList) throws Except
                                                         aggregatedCommitInfoSerializer.deserialize(
                                                                 bytes)))
                         .collect(Collectors.toList());
-        aggregatedCommitter.commit(aggregatedCommitInfos);
+        List<AggregatedCommitInfoT> commit =
+                aggregatedCommitter.restoreCommit(aggregatedCommitInfos);
+        if (CollectionUtils.isNotEmpty(commit)) {
+            log.error("aggregated committer error: {}", commit.size());
+            throw new CheckpointException(CheckpointCloseReason.AGGREGATE_COMMIT_ERROR);
+        }
```

Comment:
We will commit/abort the commit info in restoreCommit, and this method will return the commit info which the commit failed.

So when the commitInfo restore fails, then the job will fail. I am not sure if this will cause the job cannot restart? If a commit info is timeout then it cannot commit success.